### ISSUE TITLE
nerd-icons-faicon doesn't work with `concat` in MS Windows Emacs

### ISSUE THREAD

**Issue by AnweshGangula** [rainstormstudio/nerd-icons.el#85]
I'm using the below command to create a string with color-pallete icon which I want to use along with colorful-mode. But I'm not able to use the `nerd-icons-faicon` function with `concat`. It works fine with the `insert` function but not with `concat`.

I posted about this in colorful-mode as well - which you can find [here](https://github.com/DevelopmentCool2449/colorful-mode/issues/3) more details.

But as mentioned in the issue mentioned above:

## Problem 
I can see that when I execute this command (`C-c C-c`) with `insert` I get the icon properly, but when I execute with `concat` the result is not rendered properly

### using `result`
```lisp
(insert (nerd-icons-faicon "nf-fae-palette_color") " ")
```
![result command output](https://github.com/rainstormstudio/nerd-icons.el/assets/22996896/4597d8f8-100f-4cec-8138-ab14575ee08e)

### using `concat`

```lisp
(concat (nerd-icons-faicon "nf-fae-palette_color") " ")
```
![concat result](https://github.com/rainstormstudio/nerd-icons.el/assets/22996896/5e193a4c-548f-4c42-b8fd-07c233ab8d2b)

## Additional details

I also asked for what the issue could be in this [superuser post](https://superuser.com/a/1846942/863212) and found that the box that is currently being rendered with `concat` is in the Private Use Area. Does this mean this particular icon needs a specific font family?

**Comment by seagle0128** [comment]
The root cause is org tangle has difference face for the results. The face is not correct to display nerd icons, since no font specified. You can use `describe-char` to check. Actually `(insert (concat (nerd-icons-faicon "nf-fae-palette_color") " "))` will display the icons correctly.

**Comment by AnweshGangula** [comment]
Yeah, this is working fine when I use `insert`, But I need to be able to se `concat` so that I can call `(setq colorful-prefix-string (concat (nerd-icons-faicon "nf-fae-palette_color") " "))` in my init file to customize the `colorful-mode` face-attribute.

**Comment by dr-scsi** [comment]
@AnweshGangula - Maybe I'm missing the point, but this is what I get on Windows:
![nerd](https://github.com/rainstormstudio/nerd-icons.el/assets/59777257/a19ffd74-6526-4610-b607-d31362da21c6)

This is with Emacs 30.0.50 (cb67c34bcf) built with MSYS2/MinGW.

**Comment by AnweshGangula** [comment]
@dr-scsi, yes it's working fine when I use `insert`, but if I use just `cocat` without `insert` in a codeblock and execute it, I don't see the icon (as mentioned this [screenshot above](https://private-user-images.githubusercontent.com/22996896/342703499-5e193a4c-548f-4c42-b8fd-07c233ab8d2b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk3NTQyNDksIm5iZiI6MTcxOTc1Mzk0OSwicGF0aCI6Ii8yMjk5Njg5Ni8zNDI3MDM0OTktNWUxOTNhNGMtNTQ4Zi00YzQyLWI4ZmQtMDdjMjMzYWI4ZDJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjMwVDEzMjU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgzMjQ3NTMxYmQ0NTc5NzU2YTYxNjU4ZmYxYjA5NGZmMzIzMDMyODM0Yjc0M2ZjYzBhZTMzYWMzNmMwNDE0NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pvmDcwSn9Mbd0bhZQkaG92_Opm1HzNqzC_4pbZrnn9s)
